src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/SendDailyReportTask.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.hmcts.reform.blobrouter.tasks;
+
+import net.javacrumbs.shedlock.spring.annotation.SchedulerLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.blobrouter.model.out.EnvelopeSummaryItem;
+import uk.gov.hmcts.reform.blobrouter.services.email.EmailSender;
+import uk.gov.hmcts.reform.blobrouter.services.report.ReportCsvWriter;
+import uk.gov.hmcts.reform.blobrouter.services.report.ReportService;
+
+import java.io.File;
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+@Component
+@ConditionalOnExpression("!'${spring.mail.host}'.equals(null) "
+    + "&& !'${spring.mail.host}'.equals('false') "
+    + "&& ${scheduling.task.send-daily-report.enabled:true}")
```

Comment:
let's just check the `enabled` flag (like in other scheduled jobs) if possible.

Reply:
@rkondratowicz Done